frontend/src/components/Notification/Notification.tsx

```diff
@@ -1,18 +1,84 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import Popup from 'reactjs-popup';
+import { useReq } from '../../context/req';
+import subscribeUser from './subscribeUser';
 import styles from './notification.module.css';
+import 'reactjs-popup/dist/index.css';
 import { notificationBadge, notificationBell } from '../../icons/other';
 
-interface NotificationProps {
-  isNotified?: boolean
-}
-
-const Notification = ({ isNotified }: NotificationProps) => (
-  <div className={styles.container}>
-    <img src={notificationBell} className={styles.bell} alt="notification bell icon" />
-    {isNotified && (
-      <img src={notificationBadge} className={styles.badge} alt="notification badge icon" />
-    )}
-  </div>
-);
+type Message = {
+  time: Date;
+  title: string;
+  body: string;
+};
+
+const Notification = () => {
```

Comment:
the notification components is dismounted and remounted on page change, which resubscribes the user. ideally wed want them to only subscribe once, would it be possible to either create a shared header component across routes, or move notification state to a context?

Reply:
Right, this is pretty important. I'll implement this in a different PR.